### ISSUE TITLE
ReferenceError: TextEncoder is not defined

### ISSUE THREAD

**Issue by DuckHunter213** [Automattic/mongoose#10638]
**Do you want to request a *feature* or report a *bug*?**

Bug

**What is the current behavior?**

after install mongoose show me a error 

```
const utf8Encoder = new TextEncoder();
                    ^

ReferenceError: TextEncoder is not defined
```

**If the current behavior is a bug, please provide the steps to reproduce.**

```
in a clean installation run the following commands

npm i mongoose
node getting-started.js
```

```
// getting-started.js
const mongoose = require('mongoose');

main().catch(err => console.log(err));

async function main() {
  await mongoose.connect('mongodb://localhost:27017/test');
}
```

**What is the expected behavior?**

show the error 
```
const utf8Encoder = new TextEncoder();
                    ^

ReferenceError: TextEncoder is not defined
```

**What are the versions of Node.js, Mongoose and MongoDB you are using? Note that "latest" is not a version.**
```
node --version
v10.19.0
```

```
mongo --version
MongoDB shell version v5.0.2
Build Info: {
    "version": "5.0.2",
    "gitVersion": "6d9ec525e78465dcecadcff99cce953d380fedc8",
    "openSSLVersion": "OpenSSL 1.1.1f  31 Mar 2020",
    "modules": [],
    "allocator": "tcmalloc",
    "environment": {
        "distmod": "ubuntu2004",
        "distarch": "x86_64",
        "target_arch": "x86_64"
    }
}
```
```
cat package.json 
{
  "dependencies": {
    "express": "^4.17.1",
    "jsonwebtoken": "^8.5.1",
    "mongoose": "^6.0.2",
    "nodemon": "^2.0.12",
    "utf8": "^3.0.0",
    "whatwg-url": "^9.0.0"
  },
  "scripts": {
    "start": "node index.js",
    "rnodemon": "nodemon index.js"
  }
}
```

**Comment by Kamikadze4GAME** [comment]
Hello @DuckHunter213 

Please provide full stack trace of the error

**Comment by rou9e** [comment]
Hey,

Having the same issue.

```
node_modules/whatwg-url/dist/encoding.js:2
const utf8Encoder = new TextEncoder();

ReferenceError: TextEncoder is not defined
	at Object.<anonymous> (node_modules/whatwg-url/dist/encoding.js:2:21)
	at Module._compile (internal/modules/cjs/loader.js:778:30)
	at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)
	at Module.load (internal/modules/cjs/loader.js:653:32)
	at tryModuleLoad (internal/modules/cjs/loader.js:593:12)
	at Function.Module._load (internal/modules/cjs/loader.js:585:3)
	at Module.require (internal/modules/cjs/loader.js:692:17)
	at require (internal/modules/cjs/helpers.js:25:18)
	at Object.<anonymous> (node_modules/whatwg-url/dist/url-state-machine.js:5:34)
	at Module._compile (internal/modules/cjs/loader.js:778:30)
```

**Comment by DuJiming** [comment]
same

**Comment by Kamikadze4GAME** [comment]
You use NodeJS v10.x

Mongoose v6.x supports NodeJS >= v12.0.0 (See https://mongoosejs.com/docs/migrating_to_6.html#version-requirements).

**Comment by jimmywarting** [comment]
you should just update nodejs

**Comment by vkarpov15** [comment]
@Kamikadze4GAME is correct. Mongoose 6 requires Node 12 or higher, so this is expected behavior. Mongoose 6 does not support Node 10.

**Comment by matthlavacka** [comment]
I get the same issue on 12.21.0 🤔 

```
meteor node -v
v12.21.0
```

**Comment by vkarpov15** [comment]
@matthlavacka can you please provide a script or repo that demonstrates the issue you're seeing?

FWIW, this issue comes from the npm package whatwg-url, which is an upstream dep of the mongodb node driver:

```
mongoose@6.0.3
└─┬ mongodb@4.1.1
  └─┬ mongodb-connection-string-url@2.0.0
    └── whatwg-url@9.1.0
```

**Comment by matthlavacka** [comment]
Sorry for the false alarm, it was an issue with `jest`. Although not sure why this update brings it up.

This fixed my issue if someone stumbles upon the same problem: https://stackoverflow.com/questions/57712235/referenceerror-textencoder-is-not-defined-when-running-react-scripts-test

**Comment by vkarpov15** [comment]
Please note that we strongly recommend **not** using Jest's jsdom test environment to run tests. Here's why: https://mongoosejs.com/docs/jest.html

We also strongly recommend using a different test framework, Jest is an antipattern.

**Comment by SeifHafri** [comment]
the issue seems to be solved after updating from 10 to 16.8.0
